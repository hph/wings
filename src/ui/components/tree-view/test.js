import React from 'react';
import renderer from 'react-test-renderer';

import { TreeView } from './index';

jest.mock('../recursive-inner-tree', () => 'RecursiveInnerTree');

const defaultProps = {
  config: {
    cwd: '~/Code',
  },
  dispatch: () => {},
};

const createSnapshot = (props = defaultProps) => {
  expect(
    renderer.create(<TreeView {...props} />).toJSON(),
  ).toMatchSnapshot();
};

test('TreeView renders a basic wrapper around RecursiveInnerTree', () => {
  createSnapshot();
});
