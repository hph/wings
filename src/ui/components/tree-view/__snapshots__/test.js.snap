// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeView does not have a postfix in the / directory 1`] = `
<div
  className="root"
>
  <div
    className="path"
    title="/"
  >
    /
  </div>
  <div
    className="scrollWrapper"
  >
    <RecursiveInnerTree
      dispatch={[Function]}
      path="/"
    />
  </div>
</div>
`;

exports[`TreeView renders a basic wrapper around RecursiveInnerTree 1`] = `
<div
  className="root"
>
  <div
    className="path"
    title="~/Code/"
  >
    ~/Code/
  </div>
  <div
    className="scrollWrapper"
  >
    <RecursiveInnerTree
      dispatch={[Function]}
      path="~/Code"
    />
  </div>
</div>
`;
